File: src/src_core/fmg/core/img/MosaicsGroupImg.cs
using System;
using System.Collections.Generic;
using fmg.core.types;
using fmg.common.geom;
using fmg.common.geom.util;

namespace fmg.core.img {

   /// <summary> Representable <see cref="EMosaicGroup"/> as image </summary>
   /// <typeparam name="TImage">plaform specific image</typeparam>
   public abstract class MosaicsGroupImg<TImage> : PolarLightsImg<EMosaicGroup, TImage>
      where TImage : class
   {
      protected MosaicsGroupImg(EMosaicGroup group) : base(group) {}

      public EMosaicGroup MosaicGroup => Entity;

      protected IEnumerable<PointDouble> GetCoords() {
         double sq = Math.Min( // size inner square
            Width - Padding.LeftAndRight,
            Height - Padding.TopAndBottom);
         var vertices = 3 + MosaicGroup.Ordinal(); // vertices count
         var center = new PointDouble(Width / 2.0, Height / 2.0);
         return (MosaicGroup != EMosaicGroup.eOthers)
            ? FigureHelper.GetRegularPolygonCoords(vertices, sq/2, center, RotateAngle)
            : FigureHelper.GetRegularStarCoords(4, sq/2, sq/5, center, RotateAngle);
      }

   }

}

File: src/src_core/fmg/core/img/MosaicsGroupImg.java
package fmg.core.img;

import java.util.stream.Stream;

import fmg.common.geom.PointDouble;
import fmg.common.geom.util.FigureHelper;
import fmg.core.types.EMosaicGroup;

/**
 * Representable {@link fmg.core.types.EMosaicGroup} as image
 *
 * @param <TImage> plaform specific image
 */
public abstract class MosaicsGroupImg<TImage> extends PolarLightsImg<EMosaicGroup, TImage> {

   public MosaicsGroupImg(EMosaicGroup group) { super(group); }
   public MosaicsGroupImg(EMosaicGroup group, int widthAndHeight) { super(group, widthAndHeight); }
   public MosaicsGroupImg(EMosaicGroup group, int widthAndHeight, Integer padding) { super(group, widthAndHeight, padding); }

   public EMosaicGroup getMosaicGroup() { return getEntity(); }
   public void setMosaicGroup(EMosaicGroup value) { setEntity(value); }

   protected Stream<PointDouble> getCoords() {
      double sq = Math.min( // size inner square
            getWidth()  - getPadding().getLeftAndRight(),
            getHeight() - getPadding().getTopAndBottom());
      int vertices = 3 + getMosaicGroup().ordinal(); // vertices count
      Stream<PointDouble> points = (getMosaicGroup() != EMosaicGroup.eOthers)
            ? FigureHelper.getRegularPolygonCoords(vertices, sq/2, getRotateAngle())
            : FigureHelper.getRegularStarCoords(4, sq/2, sq/5, getRotateAngle());

      // adding offset
      double offsetX = getWidth() / 2.0;
      double offsetY = getHeight() / 2.0;
      return points.map(p -> {
         p.x += offsetX;
         p.y += offsetY;
         return p;
      });
   }

}
